include/dlaf/eigensolver/get_red2band_panel_nworkers.h

```diff
@@ -0,0 +1,27 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+#pragma once
+
+#include <pika/runtime.hpp>
+
+#include "dlaf/common/assert.h"
+#include "dlaf/tune.h"
+
+namespace dlaf::eigensolver::internal {
+
+inline size_t getReductionToBandPanelNWorkers() noexcept {
+  const size_t nworkers = getTuneParameters().red2band_panel_nworkers;
+
+  DLAF_ASSERT(nworkers >= 1 && nworkers < pika::get_num_worker_threads(), nworkers);
```

Comment:
@msimberg @rasolca not sure about the actual requirements. is it better to leave one for pika runtime so that other stuff can still be scheduled?

Reply:
Algorithmically I don't know, but with the yielding pika barrier I don't see a reason to leave threads unused. However, I guess this should actually be `pika::resource::get_pool("default").get_os_thread_count()`, the MPI pool thread is included in `pika::get_num_worker_threads()`?

Reply:
> Algorithmically I don't know, but with the yielding pika barrier I don't see a reason to leave threads unused. 

Ok.

> I guess this should actually be pika::resource::get_pool("default").get_os_thread_count()

It's a good point. Both here and in the default value setup we want to know the amount of available threads, not considering MPI dedicated thread.

> the MPI pool thread is included in pika::get_num_worker_threads()?

I don't know, I have to check.

Reply:
> I don't know, I have to check.

Sorry @albestro, the parentheses were not a question but a statement. It does include the MPI pool (it's essentially the same as what you pass to `--pika:threads`, or what's inferred for it internally, which includes all pools).

Reply:
> is it better to leave one for pika runtime so that other stuff can still be scheduled?

I'm coming back to this again, and indeed it might be a good idea to do `max(num_threads - 1, 1)`. Whether it's needed for sure will probably depend on the algorithm and need profiling, but it's most likely quite a safe choice to leave one thread out.